kratos/linear_solvers/amgcl_ns_solver.h

```diff
@@ -408,7 +411,7 @@ class AMGCL_NS_Solver : public LinearSolver< TSparseSpaceType,
             const unsigned int eq_id = it->EquationId();
             if( eq_id < rA.size1() )
             {
-                mp[eq_id]  = (it->GetVariable().Key() == PRESSURE);
+                mp[eq_id]  = (it->GetVariable().Key() == *mpPressureVar);
```

Comment:
Could you use `it->GetVariable().Name()` instead of key and leave the setting as a string?. This way you don't need to access the components and you can use any variable type.


If you did this to force `Variable<double>`ignore the comment, I have not much experience with this part of Kratos.

Reply:
yes, this code only makes sense if the variable is `Variable<double>` as this variable has to somehow equivalent to `PRESSURE` (so for example we use `SCALED_PRESSURE`) , so maybe we can leave it as this...